### DISCUSSION TITLE
Is there any way to immediately upload images after picking it from camera or gallary?

### DISCUSSION THREAD

**Discussion by rgtstha** [flutter-form-builder-ecosystem/form_builder_image_picker#46] [Q&A]
I want a feature like immediately uploading images to the server when the user picks images and shows the upload progress in the UI. After uploading is completed, it should replace the form value with the image URL got from the server. And disallow the user to submit the form until the upload process is completed. 

I know there is `onChanged()` method but this is not working perfectly for me. Have any done this previously with `formBuilder`?

**Comment by deandreamatias** [comment]
Hi!
I think that is a 4 different steps that you want:

1. Immediately uploading images to the server when the user picks images
A: Here you need use the `onChanged` method to send the pictures to server, with your own state management. Maybe you
2. shows the upload progress in the UI
A: Maybe you can use the property `loadingWidget` to show a custom widget when upload your images. Or, build your own in other place of your view
3. Replace the form value with the image URL
A: You can use the field form key or the form key to update the field value with a image. Take a look on examples on [this readme](https://github.com/flutter-form-builder-ecosystem/flutter_form_builder#programmatically-changing-field-value)
4. disallow the user to submit the form until the upload process
A: To this, you need use again your state management to switch the value `enable` on image picker field

**Reply by rgtstha** [reply]
is there any plan to integrate this into the package?

**Reply by deandreamatias** [reply]
What point do you refer?
